js/gameboard.js

```diff
@@ -1,110 +1,160 @@
-  var context = target.getContext("2d");
-  // shape that your drawing
-  context.beginPath();
-  var radius = 5; // the size is the radius in pixels    
-  var start = 0; 
-  var end = Math.PI*2;
-  context.arc(x, y, radius, start, end);
-  //calling the varible to the colors
-  context.fillStyle = color;
-  // filling the circles 
-  context.fill();
+
+/**
+ * This is the canvase class that is just a wrapper/refactor of the previous code base.
+ */
+function canvas() {
+  // the canvas element on the web page
+  this.canvasContext = $('#gameboard').get(0).getContext("2d")
+  
+  // the canvas's element offset
+  this._canvasOffet = $('#gameboard').offset();
+  
+  // the image being given by google MAP API in the href
+  this.imageUrl = strip(location.search.slice(1).split('screenshot')[1], '%22').slice(1)
+
+  // the color the pain brush is using
+  this.color = '';
+  
+  // some url
+  this.brickwall = "https://s-media-cache-ak0.pinimg.com/736x/94/aa/66/94aa6684bdc152457d6983e58fd24bd2.jpg";
+  
+  /**
+   * This renders the image into the canvas and sets the
+   * hieght and with of the image.
+   */
+  this.imageInput = function() {
+    // new image
+    var img = new Image();
+    var context = this.canvasContext;
+
+    // upon image loading 
+    img.onload = function () {
+      // positioning and draw the image 
+      context.drawImage(img, 0, 0);
+    };
+    // setting the empty image to an image 
+    img.src = this.imageUrl;
+    //prevents the image from stretching out 
+    img.width = "640px";
+    img.height = "480px";
+  }
+  
+  /**
+   * This chnages the color of the paint brush
+   * @param {string} color The color choosen
+   */
+  this.setColor = function(color) {
+    switch (color) {
+      case 'green':
+        this.color = "#009933";
+        break;
+      case 'red':
+        this.color = "#ff0000";
+        break;
+      case 'blue':
+        this.color = "#0000ff";
+        break;
+      case 'yellow':
+        this.color = "#ffff00";
+        break;
+      default:
+        if (!color) {
+          throw new Error('color given was undefined.')
+          return;
+        } else {
+          console.log('color does not exist')
+          return;
+        }
+    }
+  }
+
+  /**
+   * This this takes the the current clinet mouse x and y coordinates on the web page
+   * and gets the offset in order to draw to the canvas.
+   * This calcuates the radius of the brashes dialtion in size.
+   * @param {number} mouseX mouse x position
+   * @param {number} mouseY mouse y position
+   */
+  this.draw = function(mouseX, mouseY) {
+    var x = mouseX - this._canvasOffet.left;
+    var y = mouseY - this._canvasOffet.top;
+    var end = Math.PI * 2;
+    var radius = 5; // the size is the radius in pixels    
+    var start = 0;
+
+    // if x & y doesnt exist, get out of the function
+    if (!x || !y) {
+      console.log("x: ", x, "y: ", y);
+      return false;
+    }
+
+    // shape that your drawing
```

Comment:
what**

Reply:
@andreGarvin on line 132 you have "shape that your drawing" unless you mean shape what you're drawing?